src/main/kotlin/racingCar/view/InputView.kt

```diff
@@ -11,9 +11,8 @@ class InputView {
         return input?.split(",")
     }
 
-    fun readTryCount(): String? {
-        println("시도회수를 입력해주세요")
-        return readLine()
+    fun readTryCount(): Int? {
+        println("시도 횟수를 입력해주세요")
+        return readLine()?.toInt()
     }
```

Comment:
코틀린의 null 때문에 고생을 많이 하신 것 같네요 😂
console에서 입력 받는 경우 저는 내부에서 null 처리를 해주는데요, 
바로 코틀린에는 [smart cast](https://kotlinlang.org/docs/typecasts.html#smart-casts) 라는 것이 있기 떄문입니다!

슬랙에서 보니 null 처리에 관한 것도 공부하셨던데 적용해 보시면 좋을 것 같습니다 ㅎㅎ

```suggestion
    fun readTryCount(): Int {
        println("시도 횟수를 입력해주세요")
        val input = readLine() ?: throw IllegalArgumentException()
        return input.toInt()
    }
```

Reply:
남겨주신 자료 잘 봤습니다 정말 코틀린 점점 사랑하게 되네요...❤️
그런데 `Suggested change` 해주신 코드에서 어떤 부분이 smart cast 와 관련이 있는지는 아직 감이 잘 안오네요
혹시 어떤 부분이 관련된 부분일까요?

Reply:
추가적으로 백엔드 크루 주노의 피드백으로 `readln()` 메서드가 NPE 을 방지해주는 메서드라고 알려줬습니다!
해당 메서드 사용해서 리팩터링 하겠습니다.

Reply:
smart cast는 null이 될 수 있는 값을 처리하는 기능이 있습니다.
null에 대한 검사를 한 뒤의  코드는 확실하게 null이 될 수 없다는 정보를 가지고 있기에,
null이 될 수 있는 값을 null이 될 수 없는 값으로 타입 변환을 해줍니다.

그래서 위의 코드에서 21번째 줄에서 null인 경우 exception를 던진후, input은 null 이 아니라는 확신이 있기 때문에 
`readTryCount`의 반환값을 nullable한 값이 아닌 non-nullable 값으로 반환할 수 있게 됩니다 !